src/box/xrow_update_route.c

```diff
@@ -260,15 +260,16 @@ xrow_update_route_branch(struct xrow_update_field *field,
 		assert(child.type == XUPDATE_BAR);
 		child.bar.path += path_offset;
 		child.bar.path_len -= path_offset;
+		/* Turn leaf scalar bar to scalar. */
+		if (child.bar.path_len == 0 &&
+		    xrow_update_op_is_scalar(child.bar.op)) {
+			struct xrow_update_scalar s = child.bar.scalar;
+			child.type = XUPDATE_SCALAR;
+			child.scalar = s;
+		}
 		/*
-		 * Yeah, bar length can become 0 here, but it is
-		 * ok as long as it is a scalar operation (not '!'
-		 * and not '#'). When a bar is scalar, it operates
-		 * on one concrete field and works even if its
-		 * path len is 0. Talking of '#' and '!' - they
-		 * are handled by array and map 'branchers'
-		 * internally, below. They reapply such
-		 * operations.
+		 * Non scalar leaf bar (bar length is 0) will be
```

Comment:
Typo? Non scalar leaf bar is when length is > 0, not 0.

Reply:
Looks correct.

Here `child.bar.path_len` is referenced as length. This length can be 0 for scalar and non-scalar leafs. The branch above handles scalar so this comment is for non-scalar leaf. (I rephrased the previous comment as scalar leaf is now handled explicitly.) 

May be I'd better put () right after leaf word.

Reply:
Changed to `(path length is 0)`.